src/config.rs

```diff
@@ -273,6 +303,16 @@ fn parse_default<T: FromStr>(env: &str, default: T) -> Result<T, Error> {
     parse(env).map(|v| v.unwrap_or(default))
 }
 
+fn parse_duration(env: &str) -> Result<Option<Duration>, Error> {
+    parse::<String>(env)?
+        .map(|ds| duration_str::parse(&ds).map_err(|_| Error::EnvVar(env.to_string(), ds)))
+        .transpose()
+}
+
+fn parse_duration_default(env: &str, default: Duration) -> Result<Duration, Error> {
+    parse_duration(env).map(|v| v.unwrap_or(default))
```

Comment:
why do we need to map this? seems like we could just
```suggestion
    parse_duration(env).unwrap_or(default)
```

Reply:
parse_duration returns Result<Option<Duration>>, we want a Result<Duration>. We default if its unset, not if it is invalid

Reply:
ah, yup. I missed the result wrapping the option. ty